src/ggml.c

```diff
@@ -1681,6 +1681,7 @@ void * ggml_new_buffer(struct ggml_context * ctx, size_t nbytes) {
 }
 
 struct ggml_tensor * ggml_dup_tensor(struct ggml_context * ctx, const struct ggml_tensor * src) {
+    GGML_ASSERT(src);
```

Comment:
Do not check for NULL pointers with `GGML_ASSERT`, there are way too many cases and it would bloat massively the code to add so many checks. If necessary use `assert` instead so that it is only enabled in debug builds.

Reply:
the simple C assert() is not that useful but better than nothing. Done.

Reply:
I would also expect some consistency though, what the reason for only randomly checking for `NULL` pointers here? Nearly every ggml function takes a pointer.